### ISSUE TITLE
[source-mixpanel] cohorts members are missing members

### ISSUE THREAD

**Issue by RobinHerzog** [airbytehq/airbyte#37833]
### Connector Name

mixpanel

### Connector Version

2.2.0

### What step the error happened?

During the sync

### Relevant information

The cohort members are not extracting all members related to a cohort.
I think the issue comes form the fact that an member / profile can be in multiple cohorts at the same time and the unique key is based on $distinct_id and not $distinct_id + cohort_id

### Contribute

- [ ] Yes, I want to contribute

**Comment by natikgadzhi** [comment]
You mean you suspect that since a member's PK is set to be `$distinct_id`, when the same member appears in another cohort, it gets deduped out in the destination?

I _think_ we should be able to do a composite PK like you propose. I don't remember if you could do that in the Connector Builder easily, but it could be a one-line PR. Would you be interested in pushing this up so I can run the CI against it?

**Comment by RobinHerzog** [comment]
I ran the Mixpanel exports within my own Mixpanel dataset. I have many cohorts (> 10) and 90.000 profiles.
I then ran a count(*) + group by cohort_id and I confirm that the number were very far from the what Mixpanel announces as count or the table cohorts.

I did a test in the no-code builder to run a cohort member extractor with a composite key: cohort_id + $distinct_id. This worked.

@natikgadzhi I'll try to make the push, I never did a PR on airbyte code
